File: nosh_now_application/lib/data/models/food.dart
import 'package:nosh_now_application/data/models/merchant.dart';

class Food {
  int foodId;
  String foodName;
  String foodImage;
  String foodDescribe;
  double price;
  int status;
  Merchant? merchant;
  Food(
      {required this.foodId,
      required this.foodName,
      required this.foodImage,
      required this.foodDescribe,
      required this.price,
      required this.status,
      this.merchant});
  factory Food.fromJson(Map<dynamic, dynamic> json) {
    return Food(
      foodId: json['id'],
      foodName: json['foodName'],
      foodImage: json['foodImage'],
      foodDescribe: json['foodDescribe'],
      price: json['price'] / 1.0,
      status: json['status'],
      merchant:
          json['merchant'] != null ? Merchant.fromJson(json['merchant']) : null,
    );
  }
}

File: nosh_now_apis/Models/Food.cs
namespace MyApp.Models
{
    public class Food : IModel
    {
        public int Id { get; set; }
        public string FoodName { get; set; }
        public byte[] FoodImage {get; set; }
        public string FoodDescribe { get; set; }
        public double Price { get; set; }
        public int Status { get; set; } // 1 is sell, 2 is sold out, 3 is stop sell
        public int MerchantId { get; set; }
        public virtual Merchant Merchant { get; set; }
        public virtual ICollection<OrderDetail> OrderDetails{ get; set; }
    }   
}